keycloak/client.go

```diff
@@ -286,6 +294,21 @@ func (c Client) ListGroups(ctx context.Context) ([]Group, error) {
 		return nil, err
 	}
 
+	for _, g := range groups {
+		logger := log.FromContext(ctx)
+		subgroups, err := c.getChildGroups(ctx, token, *g.ID)
+		if err != nil {
+			apiErr, ok := err.(*gocloak.APIError)
+			if ok && apiErr.Code == 404 {
+				logger.Info("Could not fetch sub groups with error 404 - assuming Keycloak 22 API", "error", err, "group", g.Name)
+			} else {
+				return nil, fmt.Errorf("failed to fetch sub groups: %w", err)
+			}
+		} else {
+			g.SubGroups = &subgroups
+		}
```

Comment:
Very nit-picky: This make the code much more easier to understand for me since the log and the "we just use subgroups" are grouped closer together.

```suggestion
		if err != nil {
			apiErr, ok := err.(*gocloak.APIError)
			if ok && apiErr.Code == 404 {
				return nil, fmt.Errorf("failed to fetch sub groups: %w", err)
			}
			logger.Info("Could not fetch sub groups with error 404 - assuming Keycloak 22 API", "error", err, "group", g.Name)
		} else {
			g.SubGroups = &subgroups
		}
```